kakarot_scripts/ef_tests/fetch.py

```diff
@@ -32,20 +32,37 @@ def generate_tests():
         f"{os.path.basename(root)}_{name}": content
         for (root, _, files) in os.walk(EF_TESTS_DIR)
         for file in files
-        if file.endswith(".json") and "BlockchainTests/GeneralStateTests" in root
+        if file.endswith(".json")
+        and "BlockchainTests/GeneralStateTests" in root
+        and "Pyspecs" not in root
         for name, content in json.loads((Path(root) / file).read_text()).items()
         if content.get("network") == DEFAULT_NETWORK
     }
 
+    # Add tests from BlockchainTests/GeneralStateTests/Pyspecs
+    test_cases.update(
+        {
+            f"{name.split('::')[-1]}": content
+            for (root, _, files) in os.walk(EF_TESTS_DIR)
+            for file in files
+            if file.endswith(".json")
+            and "BlockchainTests/GeneralStateTests/Pyspecs" in root
+            for name, content in json.loads((Path(root) / file).read_text()).items()
+        }
+    )
+
     shutil.rmtree(EF_TESTS_PARSED_DIR, ignore_errors=True)
     EF_TESTS_PARSED_DIR.mkdir(parents=True, exist_ok=True)
 
     for test_name, test_case in test_cases.items():
-        json.dump(
-            test_case,
-            open(EF_TESTS_PARSED_DIR / f"{test_name}.json", "w"),
-            indent=4,
-        )
+        try:
```

Comment:
what would make this fail?

Reply:
filename too longs :(
```
[Errno 63] File name too long: 'tests/ef_tests/test_data/parsed/test_selfdestruct_created_same_block_different_tx[fork_Cancun-blockchain_test-selfdestruct_contract_address_0x6295ee1b4f6dd65047762f924ecd367c17eabf8f-entry_code_address_0xec0e71ad0a90ffe1909d27dac207f7680abba42d-call_times_1-selfdestruct_contract_initial_balance_0].json'
[Errno 63] File name too long: 'tests/ef_tests/test_data/parsed/test_selfdestruct_created_same_block_different_tx[fork_Cancun-blockchain_test-selfdestruct_contract_address_0x6295ee1b4f6dd65047762f924ecd367c17eabf8f-entry_code_address_0xec0e71ad0a90ffe1909d27dac207f7680abba42d-call_times_1-selfdestruct_contract_initial_balance_1].json'
[Errno 63] File name too long: 'tests/ef_tests/test_data/parsed/test_selfdestruct_created_same_block_different_tx[fork_Cancun-blockchain_test-selfdestruct_contract_address_0x6295ee1b4f6dd65047762f924ecd367c17eabf8f-entry_code_address_0xec0e71ad0a90ffe1909d27dac207f7680abba42d-call_times_10-selfdestruct_contract_initial_balance_0].json'
[Errno 63] File name too long: 'tests/ef_tests/test_data/parsed/test_selfdestruct_created_same_block_different_tx[fork_Cancun-blockchain_test-selfdestruct_contract_address_0x6295ee1b4f6dd65047762f924ecd367c17eabf8f-entry_code_address_0xec0e71ad0a90ffe1909d27dac207f7680abba42d-call_times_10-selfdestruct_contract_initial_balance_1].json'
```

not sure how to fix this efficiently

Reply:
given that the error is very restricted to these tests only and that we don't need to debug them I don't think it's a problem

Reply:
looks like the naming is directly produced by a `pytest.mark.parametrize` would could replace the `[]` content with numbers?

Reply:
I could theoritically but then it becomes completely useless to debug from the test name directly as if the EF-Test fails, it will fail with this parametrized name